mono_repo/lib/src/yaml.dart

```diff
@@ -0,0 +1,161 @@
+      source.trim() == source && // no leading or trailing whitespace
+      source.codeUnits.every((i) => i < 128) && // quote all non-ascii Strings
+      !_yamlSpecialStrings.contains(source.toLowerCase())) {
+    return source;
+  }
+  var output = source.replaceAllMapped(_escapeRegExp, (match) {
+    var value = match[0];
+    return _escapeMap[value] ?? _getHexLiteral(value);
+  }).replaceAll('"', r'\"');
+
+  return '"$output"';
+}
+
+bool _isSimple(Object source) =>
+    source == null || source is bool || source is num || source is String;
+
+void _writeYaml(
+    StringBuffer buffer, Object source, int indent, bool parentIsMap) {
+  final spaces = '  ' * indent;
+  if (source is String) {
+    buffer.write(_escapeString(source));
+  } else if (source == null || source is bool || source is num) {
+    buffer.write(source);
+  } else if (source is Map) {
+    if (source.isEmpty) {
+      if (parentIsMap) {
+        // Need to ensure there is a space after the map `key:`
+        buffer.write(' ');
+      }
+      buffer.write('{}');
+    } else {
+      var first = true;
+      for (var entry in source.entries) {
+        String keyLiteral;
+        if (entry.key is String) {
+          keyLiteral = _escapeString(entry.key as String);
+        } else if (_isSimple(entry.key)) {
+          keyLiteral = entry.key.toString();
+        } else {
+          throw ArgumentError('Map keys must be simple literals.');
+        }
+
+        if (first && !parentIsMap) {
+          buffer.write('$keyLiteral:');
+        } else {
+          buffer.writeln();
+          buffer.write('$spaces$keyLiteral:');
+        }
+
+        if (first) {
+          first = false;
+        }
+
+        if (_isSimple(entry.value)) {
+          buffer.write(' ');
+          _writeYaml(buffer, entry.value, 0, true);
+        } else {
+          _writeYaml(buffer, entry.value, indent + 1, true);
+        }
+      }
+    }
+  } else if (source is Iterable) {
+    if (source.isEmpty) {
+      if (parentIsMap) {
+        // Need to ensure there is a space after the map `key:`
+        buffer.write(' ');
+      }
+      buffer.write('[]');
+    } else {
+      var first = true;
+      for (var item in source) {
+        if (first) {
+          if (parentIsMap) {
+            buffer.writeln();
+          }
+          first = false;
+        } else {
+          buffer.writeln();
+        }
+        buffer.write('$spaces- ');
+        if (_isSimple(item)) {
+          _writeYaml(buffer, item, indent, false);
+        } else {
+          _writeYaml(buffer, item, indent + 1, false);
+        }
+      }
+    }
+  } else {
+    throw UnsupportedError('We do not like ${source.runtimeType}');
+  }
+}
+
+/// Strings that have special meaning in Yaml
+///
+/// Note: "yes" and "no" are not treated as `true`/`false` in pkg:yaml, but
+/// they are in many other parsers.
+const _yamlSpecialStrings = ['true', 'false', 'null', '...', 'yes', 'no'];
+
+final _yamlIndicators = [
```

Comment:
Where are these coming from, is there a canonical list somewhere you can link to? Same for _yamlSpecialStrings.

Reply:
ack